src/test/socket/listen/test_listen.rs

```diff
@@ -122,6 +125,40 @@ fn get_tests() -> Vec<test_utils::ShadowTest<(), String>> {
         }
     }
 
+    for &domain in [libc::AF_INET, libc::AF_UNIX].iter() {
+        let passing = if domain != libc::AF_UNIX {
+            set![TestEnv::Libc, TestEnv::Shadow]
+        } else {
+            set![TestEnv::Libc] // TODO: enable once we support socket() for unix sockets
+        };
+
+        for &sock_type in [libc::SOCK_STREAM, libc::SOCK_SEQPACKET].iter() {
+            for &flag in [0, libc::SOCK_NONBLOCK, libc::SOCK_CLOEXEC].iter() {
+                // add details to the test names to avoid duplicates
+                let append_args =
+                    |s| format!("{} <domain={},type={},flag={}>", s, domain, sock_type, flag);
+
+                // skip tests that use SOCK_SEQPACKET with INET sockets
+                if domain == libc::AF_INET && sock_type == libc::SOCK_SEQPACKET {
+                    continue;
+                }
+
+                tests.extend(vec![
+                    test_utils::ShadowTest::new(
+                        &append_args("test_readable"),
```

Comment:
Nit - these could use a slightly more descriptive label and name. e.g. `test_listeningSocketNotReadable`

Reply:
Changed the function names.